File: 프로그래머스/2/42586. 기능개발/기능개발.js
function solution(progresses, speeds) {
    let answer = [];
    
    // 기능 개발에 남은 날짜 구하기 
    let requireDay = progresses.map((progress, index) => {
        return Math.ceil((100 - progress) / speeds[index])
    })
    
    //
    let maxDate = requireDay[0];
    let count = 1; 
    
    for (let i=1; i<progresses.length; i++) {
        if (requireDay[i] > maxDate) {
            answer.push(count)
            maxDate = requireDay[i];
            count = 1;
        } else {
            count++;
        }
    } 
    
    answer.push(count);
    
    return answer;
}

File: 프로그래머스/2/42586. 기능개발/기능개발.py
import math

def solution(progresses, speeds):
    answer = []
    n = len(progresses)
    
    days_left = [math.ceil((100 - progresses[i])/speeds[i]) for i in range(n)]
    
    count = 0
    max_day = days_left[0]
    
    for i in range(n):
        if days_left[i] <= max_day:
            count += 1
        else:
            answer.append(count)
            count = 1
            max_day = days_left[i]
    answer.append(count)
    return answer
    
